pkg/config/controller.go

```diff
@@ -21,10 +21,23 @@ import (
 	"github.com/chaos-mesh/chaos-mesh/pkg/webhook/config/watcher"
 )
 
+// TLSConfig defines the configuration for chaos-daemon tls client
+type TLSConfig struct {
+	// ChaosDaemonClientCert is the path of chaos daemon certificate
+	ChaosDaemonClientCert string `envconfig:"CHAOS_DAEMON_CLIENT_CERT" default:""`
+	// ChaosDaemonClientKey is the path of chaos daemon certificate key
+	ChaosDaemonClientKey string `envconfig:"CHAOS_DAEMON_CLIENT_KEY" default:""`
+	// ChaosMeshCACert is the path of chaos mesh ca cert
+	ChaosMeshCACert string `envconfig:"CHAOS_MESH_CA_CERT" default:""`
+}
+
 // ChaosControllerConfig defines the configuration for Chaos Controller
 type ChaosControllerConfig struct {
 	// ChaosDaemonPort is the port which grpc server listens on
 	ChaosDaemonPort int `envconfig:"CHAOS_DAEMON_PORT" default:"31767"`
+
+	TLSConfig
```

Comment:
Can we use a pointer here?

Reply:
pointer is not good here. Because `pkg/grpc/utils` use `caCertPath`, `certPath` and `keyPath` three arguments to pass these fields, so that we have to do nil check outside the function call, which is a redundant behavior (than pass an empty value "")

But we cannot use `*TLSConfig` in `pkg/grpc/utils` directly, because depending on the `pkg/config` will trigger the `init` function and try to parse the config from env, which is not expected as a generally used package (rather than only used by controller).

Reply:
Putting the logic of "getting config from env" and the config structure leads to this problem. I don't know what's the best practice in go to parse the config . Maybe it should be split in another pkg, which obviously should be done in another PR.

Reply:
Got it